### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by a1346054**
Minor improvements fixing a few edge cases.

### CODE REVIEW

File: run-tests.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ assertEquals() {
     ((total++))
     expected=$1
     actual=$2
-    if [[ $expected = $actual ]]; then
+    if [[ $expected == "$actual" ]]; then
```
Comment (rautamiekka) [inline run-tests.sh:21]: `$expected` should be quoted too.

Comment (a1346054) [reply run-tests.sh:21]: Left hand side of bash [[ ... ]] test does not need to be quoted, however the right hand side must always be quoted if you want an exact comparison.

Comment (rautamiekka) [reply run-tests.sh:21]: Ah, ok.

**Comment by a1346054** [comment]
Rebased to fix merge conflicts
